RELEASE_PROCESS.md

```diff
@@ -0,0 +1,98 @@
+# Release Process
+
+- [Release Process](#release-process)
+  - [Breaking Changes](#breaking-changes)
+  - [Release Procedure](#release-procedure)
+    - [Creating a new release branch](#creating-a-new-release-branch)
+    - [Cutting a new release](#cutting-a-new-release)
+    - [Tagging Procedure](#tagging-procedure)
+  - [Patch release Procedure](#patch-release-procedure)
+
+This document outlines the release process for Babylon node (Babylond)
+
+Babylon follows [semantic versioning](https://semver.org), but with the following deviations to account for state-machine and API breaking changes.
+
+- State-machine breaking changes & API breaking changes will result in an increase of the minor version Y (0.Y.z).
```

Comment:
It is nice for us to keep using v0... that helps us to avoid updating dependencies / go mods with `v*`

Reply:
that is true :sweat_smile:

Reply:
Do we know why other teams are using the `v1`, `v2`, etc. method apart from image concerns?

Reply:
I did not ask anybody tbh so it is hard to tell exact specifics reasons.

Another reason I can things of though is that specifically Gaia,  uses major version to signal state breaking change, and minor version to signal api breaking one. So each `v1`, `v2` etc is kind of synchronised with upgrades and signal consensus breaking changes.  Imo this is pretty nice advantage.